packages/untp-test-suite/src/interfaces/lib/testSuiteHandler.ts

```diff
@@ -0,0 +1,310 @@
+ * each tested data file, including any warnings or errors encountered during the testing process.
+ * 
+ * Example of a test suite result:
+ * {
+ *   credentials: [
+ *     {
+ *       credentialType: "aggregationEvent",
+ *       version: "v0.0.1",
+ *       path: "/data/aggregationEvent.json",
+ *       result: "PASS"
+ *     },
+ *     {
+ *       credentialType: "conformityEvent",
+ *       version: "v0.0.1",
+ *       path: "/data/conformityEvent.json",
+ *       result: "WARN",
+ *       warnings: [
+ *         {
+ *           fieldName: "testField",
+ *           message: "This schema must NOT have additional properties"
+ *         }
+ *       ]
+ *     },
+ *     {
+ *       credentialType: "productPassport",
+ *       version: "v0.0.1",
+ *       path: "/data/productPassport.json",
+ *       result: "FAIL",
+ *       errors: [
+ *         {
+ *           fieldName: "topic",
+ *           errorType: "enum",
+ *           allowedValues: ["aggregationEvent", "conformityEvent", "productPassport"],
+ *           message: "topic field must be equal to one of the allowed values."
+ *         }
+ *       ]
+ *     }
+ *   ],
+ *   finalStatus: 'FAIL',
+ *   finalMessage: 'Your credentials are not UNTP compliant',
+ * 
+ *   // finalStatus: 'PASS',
+ *   // finalMessage: 'Your credentials are UNTP compliant',
+ * 
+ *   // finalStatus: 'WARN',
+ *   // finalMessage: 'Your credentials are UNTP compliant, but have extended the data model',
+ * }
+ * 
+ * After processing all credentials, the function generates a final report containing aggregated information
+ * about the test suite, including the overall status (PASS, WARN, or FAIL) and a corresponding message.
+ * 
+ * @example
+ * // Example usage of 'testMultiCredentialHandler' with direct credentials object:
+ * const credentialResults = await testMultiCredentialHandler({
+ *   credentials: [
+ *     {
+ *       type: "aggregationEvent",
+ *       version: "v0.0.1",
+ *       dataPath: "/data/aggregationEvent.json"
+ *     },
+ *     // Add more credentials as needed...
+ *   ]
+ * });
+ * 
+ * @example
+ * // Example usage of testMultiCredentialHandler with credentials file path:
+ * const credentialResults = await testMultiCredentialHandler('/path/to/credentials.json');
+ * 
+ * // The 'credentialResults' variable will contain the test suite result.
+ */
+
+/**
+ * Handles testing of multiple credentials against their respective schemas and generates a final report.
+ * @param {string | ConfigCredentials} credentialOrCredentialPath - The path to the credential file or the actual ConfigCredentials object.
+ * @returns {Promise<TestSuiteResult>} A promise that resolves to the test results for all credentials and the final report.
+ */
+
+export const testMultiCredentialHandler: TestMultiCredentialHandler = async (credentialOrCredentialPath: string | ConfigCredentials): Promise<ITestMultiCredentialHandlerResult> => {
```

Comment:
I am curious about what's different between this function and processTestSuite() in the core.

Reply:
Agree. All the logic handles should be in the core layer. At the interface layer, it should wrap up and make up the result to be suitable for the interface.